### ISSUE TITLE
Add Sparsemax function to libtorch

### ISSUE THREAD

**Issue by manyoso** [pytorch/pytorch#72283]
### 🚀 The feature, motivation and pitch

I would love to see sparsemax https://arxiv.org/abs/1602.02068 implemented in libtorch. There exist some python only implementations (https://github.com/KrisKorrel/sparsemax-pytorch) and (https://github.com/aced125/sparsemax) but these don't work for libtorch.

Would this be a straightforward port so that we can do a sparse softmax without having to rely upon sparse tensors and the poor performance of _sparse_softmax in the cuda implementation which requires lots of cuda stream syncs...

Thoughts?

**Comment by dagitses** [comment]
@albanD @jbschlosser would we accept a patch for this at this point?

**Comment by albanD** [comment]
I am not sure we want to add this to core at this time given the current number of citations and corresponding python implementations.
Also it should be relatively simple to translate the python implementations to c++ if you need them right now no?

**Comment by manyoso** [comment]
Yes, that's what I was planning on doing unless you wanted this in the core. It seems to be a pretty popular algorithm and thus I figured you'd want it in the core to support both python and libtorch users.

**Comment by jbschlosser** [comment]
I'd be okay with adding it. [TF has it](https://www.tensorflow.org/addons/api_docs/python/tfa/activations/sparsemax), and it's [reasonably popular on PapersWithCode](https://paperswithcode.com/method/sparsemax). Feel free to add me as reviewer if you'd like to contribute a PR.

cc @cpuhrsch in case his sparse work is relevant here

**Comment by cpuhrsch** [comment]
@manyoso - would you like this function to output a sparse Tensor (COO or CSR)?

**Comment by manyoso** [comment]
I want it to output a regular tensor not a sparse tensor at all...

**Comment by duanckham** [comment]
@cpuhrsch @jbschlosser 

Hi friends, how is your work with sparsemax going?
